src/App.jsx

```diff
@@ -27,46 +27,90 @@ function App() {
     });
   }
 
+  function getOppDirection(dir) {
+    if (dir == DIRECTION.UP) {
+      return DIRECTION.DOWN;
+    } else if (dir == DIRECTION.DOWN) {
+      return DIRECTION.UP;
+    } else if (dir == DIRECTION.RIGHT) {
+      return DIRECTION.LEFT;
+    } else {
+      return DIRECTION.RIGHT;
+    }
+  }
+
   const up = () => {
-    setCurrentDirection(DIRECTION.UP);
+      setCurrentDirection(prevDirection => {
+        if (getOppDirection(prevDirection) == DIRECTION.UP) {
+          return prevDirection
+        } else {
+          return DIRECTION.UP
+          }
+      }
+        );
   };
 
   const down = () => {
-    setCurrentDirection(DIRECTION.DOWN);
+    if ( getOppDirection() != currentDirection)
+      setCurrentDirection(prevDirection => {
+        if (getOppDirection(prevDirection) == DIRECTION.DOWN) {
+          return prevDirection
+        } else {
+          return DIRECTION.DOWN
+          }
+      }
+        );
   };
 
   const right = () => {
-    setCurrentDirection(DIRECTION.RIGHT);
+    if( getOppDirection() != currentDirection)
+      setCurrentDirection(prevDirection => {
+        if (getOppDirection(prevDirection) == DIRECTION.RIGHT) {
+          return prevDirection
+        } else {
+          return DIRECTION.RIGHT
+          }
+      }
+        );
   };
 
   const left = () => {
-    setCurrentDirection(DIRECTION.LEFT);
+    if( getOppDirection() != currentDirection)
+      setCurrentDirection(prevDirection => {
+        if (getOppDirection(prevDirection) == DIRECTION.LEFT) {
+          return prevDirection
+        } else {
+          return DIRECTION.LEFT
+          }
+      }
+        );
   };
 
   useEffect(() => {
     const timer = setInterval(moveForward.bind(this), 1 * 300);
 
     document.addEventListener("keydown", (event) => {
       const key = event.key.toLowerCase();
-      if (["w", "arrowup"].includes(key)) {
+      if (["w", "arrowup"].includes(key) ) {
```

Comment:
I will add and configure prettier for this, since you seem to use some other code formatter.

Reply:
I have setup prettier + lint-stage in shibi/master, rebase with shibi/master.

Reply:
Will run yarn run prettify ones this pr is merged to avoid confusion.